src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -262,11 +290,20 @@ public void validate() throws AppEngineException {
           "Validation Error: Java Tools jar location '"
               + JAR_LOCATIONS.get(JAVA_TOOLS_JAR) + "' is not a file.");
     }
+    try {
+      if (!isComponentInstalled("app-engine-java")) {
```

Comment:
Staging is in the process of being distributed to Maven Central. We should
be extracting ourselves out of this mess, rather than digging deeper into
it.

On Thu, Sep 8, 2016 at 12:46 PM, meltsufin <notifications@github.com> wrote:

> In src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java
> <https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/207#discussion_r78044155>
> :
>
> > @@ -262,11 +290,20 @@ public void validate() throws AppEngineException {
> >            "Validation Error: Java Tools jar location '"
> >                + JAR_LOCATIONS.get(JAVA_TOOLS_JAR) + "' is not a file.");
> >      }
> > +    try {
> > +      if (!isComponentInstalled("app-engine-java")) {
>
> It's true that the Cloud SDK reserves the right to move the files, but
> until we drop the dependency on AppCfg in the app-engine-java component in
> the Cloud SDK, we can do the file-based check without making anything more
> fragile in the library.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/207/files/29147b0c66ac9c154d990808acfebec1ffc8f716#r78044155>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HRitzbZ-DsADGWSLzsSPa2aR3TLxks5qoDvvgaJpZM4J35Q->
> .
>